File: Leetcode-Problems/Count Partitions with Even Sum Difference.cpp
class Solution
{
public:
    int countPartitions(vector<int> &nums)
    {
        int left = 0, total = 0;
        for (int i : nums)
            total += i;
        int ans = 0;
        for (int i = 0; i < nums.size() - 1; i++)
        {
            left += nums[i];
            total -= nums[i];
            if ((left - total) % 2 == 0)
                ans++;
        }
        return ans;
    }
};

File: Leetcode-Problems/Count Partitions with Even Sum Difference.java
class Solution {
    public int countPartitions(int[] nums) {
        int sum = 0, temp = 0, ans = 0;
        for (int i : nums)
            sum += i;
        for (int i = 0; i < nums.length - 1; i++) {
            temp += nums[i];
            if ((temp - (sum - temp)) % 2 == 0)
                ans++;
        }
        return ans;
    }
}